### ISSUE TITLE
enabler[] - Set up a CI/CD pipeline using GitHub Actions for automated testing and deployment to GCP.

### ISSUE THREAD

**Issue by menezmethod** [menezmethod/st-server#54]
**Description:**
As part of our efforts to streamline our development process and improve our deployment workflow, we need to set up a Continuous Integration/Continuous Deployment (CI/CD) pipeline. This pipeline will leverage GitHub Actions to automate testing and deployment of our codebase to Google Cloud Platform (GCP).

**Objectives:**
- Create GitHub Actions workflows that trigger on push and pull requests to the main branch.
- Set up automated testing within the workflow to run our suite of unit and integration tests.
- Configure deployment workflows to deploy the application to GCP upon successful tests and merge into the main branch.
- Ensure that the workflow includes steps for building the project, running tests, and handling deployment artifacts.
- Securely manage credentials and access tokens required for GCP deployment within GitHub Secrets.

**Acceptance Criteria:**
- A GitHub Actions workflow file (`.github/workflows/ci-cd.yml`) is committed to the repository.
- Automated tests are executed on every push and pull request to the main branch.
- The deployment to GCP is triggered only after successful completion of tests.
- The deployment process includes appropriate rollbacks and alerts in case of failure.
- All sensitive credentials are stored as encrypted secrets and not exposed in logs or to unauthorized users.

**Tasks:**
- [ ] Research best practices for CI/CD pipelines with GitHub Actions and GCP.
- [ ] Write the workflow configuration file for GitHub Actions.
- [ ] Set up testing scripts to run automatically within the CI pipeline.
- [ ] Configure GCP deployment specifics (e.g., service accounts, permissions, resource definitions).
- [ ] Document the setup process and any manual steps required for maintenance.

**Resources:**
- GitHub Actions documentation
- GCP deployment guides
- Access to the repository and necessary permissions on GitHub and GCP

**Risks & Mitigations:**
- Ensure that the pipeline has proper error handling to prevent broken builds from deploying.
- Regularly review and rotate GCP credentials to maintain security.
